contracts/NFTVesting/IntervalVestingNftTimeLock.sol

```diff
@@ -0,0 +1,234 @@
+            "TimeLock: vesting start time is before current time"
+        );
+
+        require(
+            address(this).balance > 0,
+            "Time:Lock: Eth should be sent to contract before initialization"
+        );
+
+        _nft = nft_;
+        _tokenId = tokenId_;
+        _nftLocker = nftLocker_;
+        _beneficiary = beneficiary_;
+        _vestingStartTime = vestingStartTime_;
+        _maxIntervals = maxInterval_;
+        _intervalDuration = intervalDuration_;
+    }
+
+    /**
+     * @dev Returns the NFT that this Timelock Contract holds.
+     */
+    function nft() public view virtual returns (IERC721) {
+        return _nft;
+    }
+
+    /**
+     * @dev Returns the token ID of the NFT being held.
+     * Returns undefined if the contract is not holding an NFT.
+     */
+    function tokenId() public view virtual returns (uint256) {
+        return _tokenId;
+    }
+
+    /**
+     * @dev Returns the NFT Locker address that will receive the NFT and ETH Discount.
+     */
+    function nftLocker() public view virtual returns (address) {
+        return _nftLocker;
+    }
+
+    /**
+     * @dev Returns the beneficiary that will receive the remaining ETH.
+     */
+    function beneficiary() public view virtual returns (address) {
+        return _beneficiary;
+    }
+
+    /**
+     * @dev Returns the time when the NFT can be released, and vesting starts in seconds since Unix epoch (i.e. Unix timestamp).
+     */
+    function vestingStartTime() public view virtual returns (uint256) {
+        return _vestingStartTime;
+    }
+
+    /**
+     * @dev Returns the maximum number of intervals set for vesting.
+     */
+    function maxIntervals() public view virtual returns (uint256) {
+        return _maxIntervals;
+    }
+
+    /**
+     * @dev Returns the set duration of each interval.
+     */
+    function intervalDuration() public view virtual returns (uint256) {
+        return _intervalDuration;
+    }
+
+    /**
+     * @dev Returns duration that NFT has been locked and vesting
+     */
+    function vestedDuration() public view returns (uint256) {
+        return block.timestamp - vestingStartTime();
+    }
+
+    /**
+     * @dev Returns the number of interval that the token has been vested for after the vesting start time.
+     */
+    function intervalsPassed() public view returns (uint256) {
+        return vestedDuration() / intervalDuration();
+    }
+
+    /**
+     * @dev Returns current vesting interval.
+     */
+    function currentInterval() public view returns (uint256) {
+        // Before vesting start time, the interval is 0.
+        if (block.timestamp < vestingStartTime()) {
+            return 0;
+        }
+
+        // After vesting start time, the interval interval count turns to 1.
+        uint256 intervals = 1 + intervalsPassed();
+
+        if (intervals > maxIntervals()) {
+            intervals = maxIntervals();
+        }
+        return intervals;
```

Comment:
Would be clearer in an if else block instead of reassigning, i.e.

```
if (intervals > maxIntervals()) {
  return maxIntervals();
} else {
  return intervals;
}
```

No need to guess on branches of code

Reply:
added thanks